File: pedidos/src/main/java/br/com/bota/ws/high/ProdutoClient.java
package br.com.bota.ws.high;

import br.com.bota.dto.ProdutoDTO;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;
import io.reactivex.Flowable;

import java.util.UUID;

@Client("${api.url.produto}")
public interface ProdutoClient {
    @Get("/{id}")
    Flowable<ProdutoDTO> buscarPorId(UUID id);
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/ws/highapi/ProdutoClient.kt
package br.com.bota.ws.highapi

import br.com.bota.lojalib.dto.ProdutoDTO
import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client
import io.reactivex.Flowable
import java.util.*

@Client("\${api.url.produto}")
interface ProdutoClient {
    @Get("/{id}")
    fun buscarPorId(  id: UUID) : Flowable<ProdutoDTO>
}